packages/components/src/theme.ts

```diff
@@ -0,0 +1,68 @@
+import Color from 'color';
+import { mapValues } from 'lodash/fp';
+import { createContext, createElement, PropsWithChildren, useContext } from 'react';
+
+type ColorPalette = {
+  /** Used as accent on header, loading spinner, background of beta version tag and some buttons */
+  primary: string;
+  /** Used as footer background */
+  secondary: string;
+  /** Used as button backgrounds, headers, links */
+  accent: string;
+  success: string;
+  warning: string;
+  danger: string;
+  /** Used as header background color, lightened for background of cells, panels, etc. */
+  light: string;
+  /** Used as text color, menu background (lightened), selected background (lightened) */
+  dark: string;
+  grey: string;
+  disabled: string;
+};
+
+export type Theme = {
+  /** Base color palette for theme. */
+  colorPalette: ColorPalette;
+};
+
+export type EnrichedTheme = Theme & {
+  /** Color mixing functions. Each takes an "intensity" and returns a modified color. */
+  colors: { [Color in keyof ColorPalette]: (intensity?: number) => string };
+};
+
+/** Add additional functionality (color mixing functions) to theme. */
+export const enrichTheme = (theme: Theme): EnrichedTheme => {
+  const colors = mapValues((color: string) => {
+    return (intensity = 1): string => {
+      return Color(color)
+        .mix(Color('white'), 1 - intensity)
+        .hex();
+    };
+  }, theme.colorPalette) as EnrichedTheme['colors'];
+
+  return {
+    ...theme,
+    colors,
+  };
+};
+
+const EnrichedThemeContext = createContext<EnrichedTheme | null>(null);
+
+type ThemeProviderProps = PropsWithChildren<{
+  theme: Theme;
+}>;
+
+/** Provides a theme to descendents via React Context. */
+export const ThemeProvider = (props: ThemeProviderProps) => {
```

Comment:
please use `const myComponent: React.FC<MyProps>`....  typing

Reply:
I still disagree with this. I think...
a) it's different than other function type definitions and it's not obvious why.
b) it provides very little benefit
c) its limitations (how to use generics with the props type?) mean that it can't be used everywhere, so (given a and b) I'd prefer to have one convention for declaring types for components (and ideally one convention for declaring types for functions).

Reply:
No need to hold up the PR on this since we want some further discussion on this with UIWG.